remill/Arch/Runtime/Operators.h

```diff
@@ -465,6 +465,111 @@ MAKE_WRITE_REF(float64_t)
 
 #undef MAKE_WRITE_REF
 
+#define MAKE_CMPXCHG(size, type_prefix, access_suffix) \
+    template <typename T> \
+    ALWAYS_INLINE static bool _CmpXchg( \
+        Memory *&memory, RnW<T> op, type_prefix ## size ## _t &expected, \
+      type_prefix ## size ## _t desired) { \
+      if (*op.val_ref == expected) {\
+        *op.val_ref = desired; \
+        return true; \
+      } else { \
+        expected = *reinterpret_cast<type_prefix ## size ## _t*>(op.val_ref); \
```

Comment:
This `reinterpret_cast` is redundant.

Reply:
It is to suppress warning about casting unsigned long to uint32_t.